apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Cloud.Spanner.V1.Logging;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+
+// ReSharper disable PrivateFieldCanBeConvertedToLocalVariable
+// ReSharper disable NotAccessedField.Local
+// ReSharper disable EmptyDestructor
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// Provides streaming access to a Spanner SQL query that automatically retries, handles
+    /// chunking and recoverable errors.
+    /// </summary>
+    public sealed class ReliableStreamReader : IDisposable
+    {
+        private AsyncServerStreamingCall<PartialResultSet> _currentCall;
+        private readonly object _sync = new object();
+        private readonly SpannerClient _spannerClient;
+        private readonly ExecuteSqlRequest _request;
+        private readonly Session _session;
+        private readonly CallSettings _callTiming;
+        private ResultSetMetadata _metadata;
+        private bool _isReading = true;
+
+        private ByteString _resumeToken;
+        private int _resumeSkipCount;
+
+        internal ReliableStreamReader(SpannerClient spannerClientImpl, ExecuteSqlRequest request, Session session)
+        {
+            _spannerClient = spannerClientImpl;
+            _request = request;
+            _session = session;
+            _clock = SpannerSettings.GetDefault().Clock ?? SystemClock.Instance;
```

Comment:
I'll run code cleanup on the file.  I think I was avoiding it because I normally run it project wide and didnt want to auto cleanup generated code (though it looks like I indeed did it by accident anyway)